.github/workflows/documentation.yml

```diff
@@ -1,33 +1,46 @@
-name: Documentation
+name: 'Documentation - Build and Deploy'
+
 on:
   workflow_dispatch:
   push:
     branches:
       - master
     paths:
+      - '.github/workflows/documentation.yml'
       - 'docs/**'
-      - 'mkdocs.yml'
   pull_request:
     paths:
+      - '.github/workflows/documentation.yml'
       - 'docs/**'
-      - 'mkdocs.yml'
+
+# Jobs will run shell commands from this subdirectory:
+defaults:
+  run:
+    working-directory: docs
 
 jobs:
   build:
-    name: build
+    if: github.event_name == 'pull_request'
     runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v2
-    - name: mkdocs build --strict
-      run: |
-        docker run --rm -v ${PWD}:/docs squidfunk/mkdocs-material build --strict --config-file docs/mkdocs.yml
+      - uses: actions/checkout@v2
+
+      - name: Build with mkdocs-material via Docker
+        run: docker run --rm -v ${PWD}:/docs squidfunk/mkdocs-material build --strict
```

Comment:
I don't get why it works without `--config-file docs/mkdocs.yml` ? 
Can you explain this? :D

Reply:
Because the `docs/mkdocs.yml` and `docs/content/` (aka `docs/` usually) is the usual structure, except for `docs/` directory being seen as `content/` which `docs_dir` in `mkdocs.yml` addresses. The workflow workdir is set to `docs/` and the Docker container mounts the `${PWD}`, instead of the root of the project. Any contributor would just do the same or volume mount `${PWD}/docs/:docs/` instead.

Reply:
ah now i get it - thanks for clarificaiton